watchman/rust/watchman_client/src/pdu.rs

```diff
@@ -559,6 +560,121 @@ pub struct SubscribeResponse {
     pub saved_state_info: Option<Value>,
 }
 
+/// The `trigger` command request.
+///
+/// The fields are explained in detail here:
+/// <https://facebook.github.io/watchman/docs/cmd/trigger#extended-syntax>
+#[derive(Deserialize, Serialize, Default, Clone, Debug)]
+pub struct TriggerRequest {
+    /// Defines the name of the trigger.
+    pub name: String,
+
+    /// Specifies the command to invoke.
+    pub command: Vec<String>,
+
+    /// It true, matching files (up to system limits) will be added to the
+    /// command's execution args.
+    #[serde(default, skip_serializing_if = "is_false")]
+    pub append_files: bool,
+
+    /// Specifies the expression used to filter candidate matches.
+    #[serde(skip_serializing_if = "Option::is_none", skip_deserializing)]
+    pub expression: Option<Expr>,
+
+    /// Configure the way `stdin` is configured for the executed trigger.
+    #[serde(
+        default,
+        skip_serializing_if = "TriggerStdinConfig::is_devnull",
+        serialize_with = "TriggerStdinConfig::serialize",
+        skip_deserializing
+    )]
+    pub stdin: TriggerStdinConfig,
+
+    /// Specifies a file to write the output stream to.  Prefix with `>` to
+    /// overwrite and `>>` to append.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stdout: Option<String>,
+
+    /// Specifies a file to write the error stream to.  Prefix with `>` to
+    /// overwrite and `>>` to append.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub stderr: Option<String>,
+
+    /// Specifies a limit on the number of files reported on stdin when stdin is
+    /// set to hold the set of matched files.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub max_files_stdin: Option<u64>,
+
+    /// Specifies the working directory that will be set prior to spawning the
+    /// process. The default is to set the working directory to the watched
+    /// root. The value of this property is a string that will be interpreted
+    /// relative to the watched root.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub chdir: Option<String>,
+}
```

Comment:
Thoughts on supporting `relative_root`?

```suggestion

    /// Since 3.4. Evaluate the trigger with respect to a path within a watched root
    #[serde(skip_serializing_if = "Option::is_none")]
    pub relative_root: Option<String>,
}
```

Reply:
added